packages/frontend-app-api/src/routing/RouteTracker.test.tsx

```diff
@@ -0,0 +1,228 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { TestApiProvider } from '@backstage/test-utils';
+import React from 'react';
+import { BackstageRouteObject } from './types';
+import { fireEvent, render } from '@testing-library/react';
+import { RouteTracker } from './RouteTracker';
+import { Link, MemoryRouter, Route, Routes } from 'react-router-dom';
+import { createPlugin } from '@backstage/core-plugin-api';
+import {
+  createRouteRef,
+  AnalyticsApi,
+  analyticsApiRef,
+} from '@backstage/frontend-plugin-api';
+import { MATCH_ALL_ROUTE } from './extractRouteInfoFromAppNode';
+
+describe('RouteTracker', () => {
+  const routeRef0 = createRouteRef();
+  const routeRef1 = createRouteRef();
+  const routeRef2 = createRouteRef();
+  const plugin0 = createPlugin({ id: 'home' });
+  const plugin1 = createPlugin({ id: 'plugin1' });
+  const plugin2 = createPlugin({ id: 'plugin2' });
+
+  const routeObjects: BackstageRouteObject[] = [
+    {
+      path: '',
+      element: <div>home page</div>,
+      routeRefs: new Set([routeRef0]),
+      plugins: new Set([plugin0]),
+      caseSensitive: false,
+      children: [MATCH_ALL_ROUTE],
+    },
+    {
+      path: '/path/:p1/:p2',
+      element: <Link to="/path2/hello">go</Link>,
+      routeRefs: new Set([routeRef1]),
+      plugins: new Set([plugin1]),
+      caseSensitive: false,
+      children: [MATCH_ALL_ROUTE],
+    },
+    {
+      path: '/path2/:param',
+      element: <div>hi there</div>,
+      routeRefs: new Set([routeRef2]),
+      plugins: new Set([plugin2]),
+      caseSensitive: false,
+      children: [MATCH_ALL_ROUTE],
+    },
+  ];
+
+  const mockedAnalytics: jest.Mocked<AnalyticsApi> = {
+    captureEvent: jest.fn(),
+  };
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should capture the navigate event on load', async () => {
+    render(
+      <MemoryRouter initialEntries={['/path/foo/bar']}>
+        <TestApiProvider apis={[[analyticsApiRef, mockedAnalytics]]}>
+          <RouteTracker routeObjects={routeObjects} />
+        </TestApiProvider>
+      </MemoryRouter>,
+    );
+
+    expect(mockedAnalytics.captureEvent).toHaveBeenCalledWith({
+      action: 'navigate',
+      attributes: {
+        p1: 'foo',
+        p2: 'bar',
+      },
+      context: {
+        extension: 'App',
+        pluginId: 'plugin1',
+        routeRef: undefined,
```

Comment:
The new route ref does not have an id, so it is always undefined. Should we remove it from the context?